src/network/networkd-dhcp4.c

```diff
@@ -443,6 +473,30 @@ static int dhcp_remove_dns_routes(Link *link, sd_dhcp_lease *lease, const struct
                 (void) route_remove(route, link, NULL);
         }
 
+        if (link_noprefixroute(link)) {
+                _cleanup_(route_freep) Route *prefix_route = NULL;
+                struct in_addr netmask;
+
+                r = route_new(&prefix_route);
+                if (r < 0)
+                        return log_link_error_errno(link, r,  "Could not allocate route: %m");
+
+                r = sd_dhcp_lease_get_netmask(lease, &netmask);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "DHCP error: No netmask: %m");
+
+                prefix_route->family = AF_INET;
+                prefix_route->dst.in.s_addr = address->s_addr & netmask.s_addr;
+                prefix_route->dst_prefixlen = in4_addr_netmask_to_prefixlen(&netmask);
+                prefix_route->prefsrc.in = *address;
+                prefix_route->scope = RT_SCOPE_LINK;
+                prefix_route->protocol = RTPROT_DHCP;
+                prefix_route->table = table;
+
```

Comment:
.. and this block are almost identical. So this could be split out in a separate function.

Though @ssahani says that the route gets deleteted automatically. But I don't understand the exact conditions when this happens. I guess it is safer to delete the route. If it turns out that this is never needed, we can drop the removal.

Reply:
No need to delete I will follow up a patch that it won't create itself.